### PR TITLE
 Bug #4710 - Desktop mode persists per-domain 

### PR DESCRIPTION
**Pull request by garvankeeley**
Todo: Tests for
- set desktop site, switch tab, ensure not set, switch back, ensure set
- navigate to another site, ensure it is off, then nav back, ensure it is on, nav again, ensure it is off
- enable private tab, set a site to desktop mode, exit private mode, verify site is not in desktop mode
- set a site to desktop mode, enable private tab, set that site to non-desktop mode, exit private mode and verify site is loaded as non-desktop

*** Note the last case reflects the way the code is currently, alternatively, that change could persist only for private browsing mode (I just thought that was a bit overkill in terms of app logic when the other way is simpler).

### CODE REVIEW

File: Client.xcodeproj/xcshareddata/xcschemes/Fennec_Enterprise_XCUITests.xcscheme [hunk 1]
```diff
@@ -173,15 +173,6 @@
       language = "en"
       region = "US"
       shouldUseLaunchSchemeArgsEnv = "NO">
-      <PreActions>
-         <ExecutionAction
-            ActionType = "Xcode.IDEStandardExecutionActionsCore.ExecutionActionType.ShellScriptAction">
-            <ActionContent
-               title = "Run Script"
-               scriptText = "/usr/local/bin/carthage checkout">
```
Comment (garvankeeley) [inline Client.xcodeproj/xcshareddata/xcschemes/Fennec_Enterprise_XCUITests.xcscheme]: Why was this here, it was slowing down running tests.

**Comment by garvankeeley** [comment]
Yes it should clear with private data, let me take a look at adding that now...

File: Client/Frontend/Browser/Tab.swift [hunk 2]
```diff
@@ -684,6 +691,56 @@ class TabWebView: WKWebView, MenuHelperInterface {
...
+                    DesktopSites.privateModeHostList.remove(host)
+                    hostList.remove(host)
+                }
+                return
+            }
+
+            if isDesktopSite, !hostList.contains(host) {
+                hostList.insert(host)
+            } else if !isDesktopSite, hostList.contains(host) {
+                hostList.remove(host)
```
Comment (farhanpatel) [inline Client/Frontend/Browser/Tab.swift]: The logic here could be simplified by doing the `!desktopSite` check first. Then we wouldn't have to do it twice below. 

`if !desktopSite { hostList.remove(host), DesktopSites.privateModeHostList.remove(host)}; return;`
